src/workflow/unlock.c

```diff
@@ -52,94 +44,63 @@ static void _finish_bip39(const char* passphrase)
     if (!keystore_unlock_bip39(passphrase)) {
         Abort("bip39 unlock failed");
     }
-    _done = true;
-}
-
-static void _mnemonic_passphrase_confirm_done(const char* passphrase)
-{
-    bool equal = STREQ(_mnemonic_passphrase_unconfirmed, passphrase);
-    util_zero(_mnemonic_passphrase_unconfirmed, sizeof(_mnemonic_passphrase_unconfirmed));
-    if (equal) {
-        _finish_bip39(passphrase);
-    } else {
-        ui_screen_stack_switch(status_create(
-            "Passphrases\ndo not match", false, STATUS_DEFAULT_DELAY, _enter_mnemonic_passphrase));
-    }
-}
-
-static void _passphrase_confirm_enter(void)
-{
-    ui_screen_stack_switch(set_password_create(_mnemonic_passphrase_confirm_done));
-}
-
-static void _mnemonic_passphrase_enter_done(const char* passphrase)
-{
-    int snprintf_result = snprintf(
-        _mnemonic_passphrase_unconfirmed,
-        sizeof(_mnemonic_passphrase_unconfirmed),
-        "%s",
-        passphrase);
-    if (snprintf_result < 0 || snprintf_result >= (int)sizeof(_mnemonic_passphrase_unconfirmed)) {
-        Abort("length mismatch");
-    }
-    ui_screen_stack_switch(
-        entry_screen_create("Confirm\nmnemonic passphrase", _passphrase_confirm_enter));
-}
-
-static void _passphrase_enter(void)
-{
-    ui_screen_stack_switch(set_password_create(_mnemonic_passphrase_enter_done));
-}
-
-static void _enter_mnemonic_passphrase(void)
-{
-    memset(_mnemonic_passphrase_unconfirmed, 0, sizeof(_mnemonic_passphrase_unconfirmed));
-    ui_screen_stack_switch(entry_screen_create("Enter\nmnemonic passphrase", _passphrase_enter));
-}
-
-static void _enter(void)
-{
-    ui_screen_stack_switch(set_password_create(workflow_unlock_enter_done));
 }
 
 static void _workflow_unlock(void)
 {
-    // "Enter password"
     ui_screen_stack_pop_all();
-    ui_screen_stack_push(entry_screen_create("Enter password", _enter));
+    char password[SET_PASSWORD_MAX_PASSWORD_LENGTH] = {0};
+    password_enter("Enter password", password);
+    workflow_unlock_enter_done(password);
+    util_zero(password, sizeof(password));
 }
 
-static void _start(void)
+static void _get_mnemonic_passphrase(char* passphrase_out)
```

Comment:
IIUC I think `mnemonic` is a bad name here.

> 
> noun
> noun: mnemonic; plural noun: mnemonics
> 
>     1.
>     a system such as a pattern of letters, ideas, or associations which assists in remembering something.

Reply:
true! let's do this in another PR as there are a lot of places to change.

Reply:
After reading bip39 I'm a little bit more OK with this name. The passphrase protects the mnemonic sentence.